pkg/graph/resolvers/plan_favorite.go

```diff
@@ -0,0 +1,74 @@
+package resolvers
+
+import (
+	"go.uber.org/zap"
+
+	"github.com/google/uuid"
+
+	"github.com/cmsgov/mint-app/pkg/authentication"
+	"github.com/cmsgov/mint-app/pkg/models"
+	"github.com/cmsgov/mint-app/pkg/storage"
+)
+
+//IsPlanFavorited checks if a model plan is a favorite.
+func IsPlanFavorited(logger *zap.Logger, principal authentication.Principal, store *storage.Store, modelPlanID uuid.UUID) (bool, error) {
+
+	favorite, err := PlanFavoriteGet(logger, principal, store, modelPlanID)
+	if err != nil {
+		return false, err
+	}
+	isFavorite := false
+	if favorite != nil {
+		isFavorite = true
+
+	}
+	return isFavorite, nil
+
+}
+
+//PlanFavoriteCreate creates a new plan favorite record in the database
+func PlanFavoriteCreate(logger *zap.Logger, principal authentication.Principal, store *storage.Store, modelPlandID uuid.UUID) (*models.PlanFavorite, error) {
+
+	favorite := models.PlanFavorite{
+		UserID: principal.ID(),
+		ModelPlanRelation: models.ModelPlanRelation{
+			ModelPlanID: modelPlandID,
+		},
+		BaseStruct: models.BaseStruct{
+			CreatedBy: principal.ID(),
+		},
+	}
```

Comment:
Extract to helper method or constructor

Reply:
We saw this a bunch across the codebase - there's actually a few instances where embedded structs are created like this, so I think it might make sense to do this across the codebase. We could still opt to do it here (just for plan favorites), but if we do, we should track the work to do it everywhere else as well. (Perhaps a new function like `NewPlanFavorite` would do well?